dotnet-test-samples/async-architectures/async-lambda-dynamodb/template.yaml

```diff
@@ -0,0 +1,160 @@
+
+Globals: # https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-specification-template-anatomy-globals.html
+    Function:
+        Timeout: 15
+        MemorySize: 256
+        Runtime: dotnet6
+        Tracing: Active # https://docs.aws.amazon.com/lambda/latest/dg/lambda-x-ray.html
+        Environment:
+            Variables:
+                DESTINATION_BUCKET: 
+                  !Sub "async-destination-${AWS::StackName}-${AWS::AccountId}"
+                RESULTS_TABLE: 
+                  !Sub "async-results-${AWS::StackName}-${AWS::AccountId}"                
+                # Powertools env vars: https://awslabs.github.io/aws-lambda-powertools-python/#environment-variables
+                LOG_LEVEL: INFO
+                POWERTOOLS_LOGGER_SAMPLE_RATE: 0.1
+                POWERTOOLS_LOGGER_LOG_EVENT: true
+                POWERTOOLS_METRICS_NAMESPACE: async-test-service
+                POWERTOOLS_SERVICE_NAME: async
+
+Resources:
+  SourceBucket:
+    Type: AWS::S3::Bucket
+    UpdateReplacePolicy: Delete
+    Properties:
+      BucketName: 
+        !Sub "async-source-${AWS::StackName}-${AWS::AccountId}"
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+        - ServerSideEncryptionByDefault:
+            SSEAlgorithm: AES256
+      PublicAccessBlockConfiguration:
+        BlockPublicAcls: true
+        BlockPublicPolicy: true
+        IgnorePublicAcls: true
+        RestrictPublicBuckets: true
+
+  DestinationBucket:
+    Type: AWS::S3::Bucket
+    UpdateReplacePolicy: Delete
+    Properties:
+      BucketName:
+        !Sub "async-destination-${AWS::StackName}-${AWS::AccountId}"
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+        - ServerSideEncryptionByDefault:
+            SSEAlgorithm: AES256
+      PublicAccessBlockConfiguration:
+        BlockPublicAcls: true
+        BlockPublicPolicy: true
+        IgnorePublicAcls: true
+        RestrictPublicBuckets: true
+
+  ToUppercaseTextTransformer:
+    Type: AWS::Serverless::Function
+    Properties:
+      FunctionName: 
+        !Sub "ToUppercaseTextTransformer-${AWS::StackName}-${AWS::AccountId}"
+      CodeUri: src/AsyncTesting.S3EventHandler
+      Handler: AsyncTesting.S3EventHandler::AsyncTesting.S3EventHandler.Function::FunctionHandler
+      Policies:
+        - S3ReadPolicy:
+            BucketName: 
+              !Sub "async-source-${AWS::StackName}-${AWS::AccountId}"
+        - S3CrudPolicy:
+            BucketName: 
+              !Sub "async-destination-${AWS::StackName}-${AWS::AccountId}"
+      Events:
+        FileUpload:
+          Type: S3
+          Properties:
+            Bucket: !Ref SourceBucket
+            Events: s3:ObjectCreated:*
+            Filter:
+              S3Key:
+                Rules:
+                  - Name: suffix
+                    Value: '.txt'
+
+  AsyncTransformTestResultsTable:
+    Type: AWS::DynamoDB::Table
+    Condition: CreateTestResources
```

Comment:
We need to discuss this, maybe it's preferable to create those as part of the test, it would be easier to maintain, and can run from the user's machine without the need to run sam deploy first